plugins/source/gcp/client/client.go

```diff
@@ -164,7 +165,25 @@ func New(ctx context.Context, logger zerolog.Logger, s specs.Source, opts source
 		}
 		c.ClientOptions = append(c.ClientOptions, option.WithCredentialsJSON(serviceAccountKeyJSON))
 	}
+	if gcpSpec.ServiceAccountImpersonation != nil {
+		c.logger.Info().Msgf("%v", gcpSpec.ServiceAccountImpersonation)
+
+		// Base credentials sourced from ADC or provided client options.
+		ts, err := impersonate.CredentialsTokenSource(ctx, impersonate.CredentialsConfig{
+			TargetPrincipal: gcpSpec.ServiceAccountImpersonation.TargetPrincipal,
+			Scopes:          gcpSpec.ServiceAccountImpersonation.Scopes,
+			// Optionally supply delegates.
+			Delegates: gcpSpec.ServiceAccountImpersonation.Delegates,
+			// Specify user to impersonate
+			Subject: gcpSpec.ServiceAccountImpersonation.Subject,
```

Comment:
These options were not documented because I could not fully understand the use case where they can be used, but are required inputs. So if a user has a need for them we can add documentation for them